internal/provider/edgegw/vpn_ipsec_resource.go

```diff
@@ -0,0 +1,522 @@
+	} else {
+		vpnTunnel, err = r.edgegw.NsxtEdgeGateway.GetIpSecVpnTunnelByName(stateRefreshed.Name.Get())
+	}
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			return nil, false, diags
+		}
+		diags.AddError("Error retrieving VPN Tunnel ID", err.Error())
+		return nil, true, diags
+	}
+
+	stateRefreshed.Description.Set(vpnTunnel.NsxtIpSecVpn.Description)
+	stateRefreshed.EdgeGatewayID.Set(r.edgegw.GetID())
+	stateRefreshed.EdgeGatewayName.Set(r.edgegw.GetName())
+	stateRefreshed.Enabled.Set(vpnTunnel.NsxtIpSecVpn.Enabled)
+	stateRefreshed.ID.Set(vpnTunnel.NsxtIpSecVpn.ID)
+	stateRefreshed.LocalIPAddress.Set(vpnTunnel.NsxtIpSecVpn.LocalEndpoint.LocalAddress)
+	stateRefreshed.LocalNetworks.Set(ctx, vpnTunnel.NsxtIpSecVpn.LocalEndpoint.LocalNetworks)
+	stateRefreshed.Name.Set(vpnTunnel.NsxtIpSecVpn.Name)
+	stateRefreshed.PreSharedKey.Set(vpnTunnel.NsxtIpSecVpn.PreSharedKey)
+	stateRefreshed.RemoteIPAddress.Set(vpnTunnel.NsxtIpSecVpn.RemoteEndpoint.RemoteAddress)
+	stateRefreshed.RemoteNetworks.Set(ctx, vpnTunnel.NsxtIpSecVpn.RemoteEndpoint.RemoteNetworks)
+	stateRefreshed.SecurityType.Set(vpnTunnel.NsxtIpSecVpn.SecurityType)
+
+	// ====== Get IPSec Profile
+	vpnTunnelSecProfile = &VPNIPSecModelSecurityProfile{}
+	secProfile, err := vpnTunnel.GetTunnelConnectionProperties()
+	if err != nil {
+		diags.AddError("Error retrieving VPN Tunnel Security Profile", err.Error())
+		return nil, true, diags
+	}
+
+	// IKE DH Groups
+	if len(secProfile.IkeConfiguration.DhGroups) > 0 {
+		vpnTunnelSecProfile.IkeDhGroups.Set(secProfile.IkeConfiguration.DhGroups[0])
+	}
+	// IKE Digest Algorithm
+	if len(secProfile.IkeConfiguration.DigestAlgorithms) > 0 {
+		vpnTunnelSecProfile.IkeDigestAlgorithm.Set(secProfile.IkeConfiguration.DigestAlgorithms[0])
+	} else {
+		vpnTunnelSecProfile.IkeDigestAlgorithm.SetNull()
+	}
+	// IKE Encryption Algorithm
+	if len(secProfile.IkeConfiguration.EncryptionAlgorithms) > 0 {
+		vpnTunnelSecProfile.IkeEncryptionAlgorithm.Set(secProfile.IkeConfiguration.EncryptionAlgorithms[0])
+	}
+	// IKE SA Lifetime
+	vpnTunnelSecProfile.IkeSaLifetime.Set(int64(*secProfile.IkeConfiguration.SaLifeTime))
+	// IKE Version
+	vpnTunnelSecProfile.IkeVersion.Set(secProfile.IkeConfiguration.IkeVersion)
+	// IKE Dead Peer Detection
+	vpnTunnelSecProfile.TunnelDfPolicy.Set(secProfile.TunnelConfiguration.DfPolicy)
+	// Tunnel DH Groups
+	if len(secProfile.TunnelConfiguration.DhGroups) > 0 {
+		vpnTunnelSecProfile.TunnelDhGroups.Set(secProfile.TunnelConfiguration.DhGroups[0])
+	}
+	// Tunnel Digest Algorithms
+	if len(secProfile.TunnelConfiguration.DigestAlgorithms) > 0 {
+		vpnTunnelSecProfile.TunnelDigestAlgorithms.Set(secProfile.TunnelConfiguration.DigestAlgorithms[0])
+	} else {
+		vpnTunnelSecProfile.TunnelDigestAlgorithms.SetNull()
+	}
+	// Tunnel Encryption Algorithm
+	if len(secProfile.TunnelConfiguration.EncryptionAlgorithms) > 0 {
+		vpnTunnelSecProfile.TunnelEncryptionAlgorithms.Set(secProfile.TunnelConfiguration.EncryptionAlgorithms[0])
+	}
```

Comment:
```go
	// Tunnel DH Groups
	if len(secProfile.TunnelConfiguration.DhGroups) > 0 {
		vpnTunnelSecProfile.TunnelDhGroups.Set(secProfile.TunnelConfiguration.DhGroups[0])
	}
	// Tunnel Digest Algorithms
	if len(secProfile.TunnelConfiguration.DigestAlgorithms) > 0 {
		vpnTunnelSecProfile.TunnelDigestAlgorithms.Set(secProfile.TunnelConfiguration.DigestAlgorithms[0])
	} else {
		vpnTunnelSecProfile.TunnelDigestAlgorithms.SetNull()
	}
	// Tunnel Encryption Algorithm
	if len(secProfile.TunnelConfiguration.EncryptionAlgorithms) > 0 {
		vpnTunnelSecProfile.TunnelEncryptionAlgorithms.Set(secProfile.TunnelConfiguration.EncryptionAlgorithms[0])
	}
```

Why TunnelDigestAlgorithms is set to Null if !> 0 while TunnelDhGroups and TunnelEncryptionAlgorithms no ?